ui/src/containers/Visualisations/CustomBarChart/Editor.js

```diff
@@ -0,0 +1,165 @@
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import { Map } from 'immutable';
+import { Tab } from 'react-toolbox/lib/tabs';
+import Tabs from 'ui/components/Material/Tabs';
+import AxesEditor from 'ui/containers/VisualiseForm/StatementsForm/AxesEditor';
+import VisualiseFilterForm from 'ui/containers/VisualiseFilterForm';
+import {
+  AddQueryButton,
+  BarChartGroupingLimitForm,
+  DescriptionForm,
+  PreviewPeriodPicker,
+  SourceViewForm,
+  StackedSwitch,
+  TimezoneForm,
+} from '../components';
+import Viewer from './Viewer';
+
+/**
+ * @param {immutable.Map} props.model - visualisation model
+ * @param {string} props.orgTimezone
+ * @param {(args: object) => void} props.updateModel
+ * @param {(args: object) => void} props.setInMetadata
+ */
+const Editor = ({
+  model,
+  orgTimezone,
+  updateModel,
+  setInMetadata,
+}) => {
+  const [tabIndex, setTabIndex] = useState(0);
+
+  return (
+    <div className="row">
+      <div className="col-md-6 left-border">
+        <div>
+          <DescriptionForm
+            visualisationId={model.get('_id')}
+            description={model.get('description', '')}
+            onChange={(description) => {
```

Comment:
We can optimise more by *using `useMemo` hooks*
```
onChangeDescription = useMemo((description) => {
  updateModel({
    schema: 'visualisation',
    id: model.get('_id'),
    path: 'description',
    value: description,
  });
}, [model.get('_id')]);

...

<DescriptionForm
  ...
  onChange={onChangeDescription}
  />
```

or *moving the code defining this handler to ui/src/containers/Visualisations/CustomBarChart/index.js
(Define handler in `compose`)

I'd like to hear others' opinions